### PR TITLE
chore(web): conversion of Web's end-to-end style auto-tests for @web/test-runner use 🏃

### PR DESCRIPTION
**Pull request by jahorton**
And now for the final set of Web automated tests - the "end to end" style ones... even if they're within a folder named `integrated`.

Converting the tests to TS gave rise to a number of smal but "fun" type errors that needed addressing.

@keymanapp-test-bot skip

### CODE REVIEW

**Review by mcdurdin — APPROVED** [review]

File: web/src/test/auto/integrated/cases/basics.spec.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  describe('Initialization', function() {
+    it('KMW should attach to the input element.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+      var singleton = document.getElementById('singleton');
+      assert.isTrue(keyman.isAttached(singleton), "KeymanWeb did not automatically attach to the element!");
+    });
+
+    it('KMW\'s initialization variable should indicate completion.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+      assert(keyman.initialized == 2, 'Keyman indicates incomplete initialization!');
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/cases/basics.spec.ts:44]: To better match the style of the other tests one could do:

```suggestion
      assert.equal(keyman.initialized, 2, 'Keyman indicates incomplete initialization!');
```

File: web/src/test/auto/integrated/cases/basics.spec.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      var divs = document.getElementsByTagName("div");
+      var match = false;
+
+      for(var i=0; i < divs.length; i++) {
+        if(divs[i] == ui.controller) {
+          match = true;
+        }
+      }
+
+      assert(match, 'Controller element has not been added to the page!');
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/cases/basics.spec.ts:89]: Could do:

```suggestion
      assert.isTrue(match, 'Controller element has not been added to the page!');
```

File: web/src/test/auto/integrated/cases/basics.spec.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      teardownKMW();
+    });
+
+    it('The Button UI initializes correctly.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+
+      // UI-module specific typings are currently not available.
+      const ui = keyman.ui as any;
+
+      assert(ui.init, 'Initialization flag is set to false!');
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/cases/basics.spec.ts:117]:
```suggestion
      assert.isTrue(ui.init, 'Initialization flag is set to false!');
```

File: web/src/test/auto/integrated/cases/basics.spec.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+      teardownKMW();
+    });
+
+    it('The Float UI initializes correctly.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+
+      // UI-module specific typings are currently not available.
+      const ui = keyman.ui as any;
+
+      assert(ui.initialized, 'Initialization flag is set to false!');
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/cases/basics.spec.ts:145]:
```suggestion
      assert.isTrue(ui.initialized, 'Initialization flag is set to false!');
```

File: web/src/test/auto/integrated/cases/basics.spec.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+      var divs = document.getElementsByTagName("div");
+      var match = false;
+
+      for(var i=0; i < divs.length; i++) {
+        if(divs[i] == ui.outerDiv) {
+          match = true;
+        }
+      }
+
+      assert(match, 'Floating controller element has not been added to the page!');
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/cases/basics.spec.ts:158]:
```suggestion
      assert.isTrue(match, 'Floating controller element has not been added to the page!');
```

File: web/src/test/auto/integrated/cases/basics.spec.ts [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+      teardownKMW();
+    });
+
+    it('The Toolbar UI initializes correctly.', function() {
+      const keyman = window['keyman'] as KeymanEngine;
+
+      // UI-module specific typings are currently not available.
+      const ui = keyman.ui as any;
+
+      assert(ui.init, 'Initialization flag is set to false!');
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/cases/basics.spec.ts:186]:
```suggestion
      assert.isTrue(ui.init, 'Initialization flag is set to false!');
```

File: web/src/test/auto/integrated/web-test-runner.config.mjs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+      }
+    })
+  ],
+  reporters: [
+    summaryReporter({}), /* local-dev mocha-style */
+    sessionStabilityReporter({}),
+    defaultReporter()
+  ],
+  /*
+    Un-comment the next two lines for easy interactive debugging; it'll launch the
```
Comment (ermshiperete) [inline web/src/test/auto/integrated/web-test-runner.config.mjs:60]: The next two or the next three lines?

**Review by ermshiperete — APPROVED** [review]

File: web/src/test/auto/integrated/cases/basics.spec.ts (hunk 1 above)
Comment (jahorton) [reply web/src/test/auto/integrated/cases/basics.spec.ts:44]: Fair enough.  I'm pretty sure this was one of the first automated tests I ever wrote for KMW, hence how rough its definition is.  We just... never polished it up.  So yeah, let's do that now.

**Review by ermshiperete — APPROVED** [review]

**Pull request merged**